upstairs/src/downstairs.rs

```diff
@@ -3673,6 +3821,589 @@ impl Downstairs {
             }
         }
     }
+
+    #[cfg(feature = "omicron-build")]
+    fn get_target_addrs(&mut self) -> Vec<SocketAddr> {
+        self.clients
+            .iter()
+            .map(|client| client.target_addr.clone())
+            .flat_map(|x| x)
+            .collect()
+    }
+
+    #[cfg(feature = "omicron-build")]
+    fn notify_nexus_of_live_repair_start(&mut self, repair: LiveRepairData) {
+        use crate::get_nexus_client;
+        use chrono::Utc;
+        use nexus_client::types::DownstairsUnderRepair;
+        use nexus_client::types::RepairStartInfo;
+        use nexus_client::types::UpstairsRepairType;
+        use omicron_uuid_kinds::GenericUuid;
+        use omicron_uuid_kinds::TypedUuid;
+        use omicron_uuid_kinds::UpstairsKind;
+        use omicron_uuid_kinds::UpstairsRepairKind;
+        use omicron_uuid_kinds::UpstairsSessionKind;
+
+        let mut repairs: Vec<DownstairsUnderRepair> =
+            Vec::with_capacity(repair.repair_downstairs.len());
```

Comment:
Personal opinion: I think explicit types add more noise than clarity in this situation
```suggestion
        let mut repairs = Vec::with_capacity(repair.repair_downstairs.len());
```

Reply:
6306218